File: solutions/go/difference-of-squares/1/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) int {
	var op int = n * ( 1 + n ) / 2
    return op * op
}

func SumOfSquares(n int) int {
	return n * (n + 1) * (2 * n + 1) / 6
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: solutions/ruby/difference-of-squares/1/difference_of_squares.rb
=begin
Write your code for the 'Difference Of Squares' exercise in this file. Make the tests in
`difference_of_squares_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/difference-of-squares` directory.
=end

class Squares
  def initialize(n)
    @numa = (1..n).to_a
  end
  
  def square_of_sum
      @numa.sum ** 2
  end
  
  def sum_of_squares
    @numa.map {|i| i ** 2}.sum
  end
  
  def difference
      square_of_sum - sum_of_squares
  end
end